templates/common/openstack/files/usr-local-bin-openstack-kubelet-nodename.yaml

```diff
@@ -66,7 +66,8 @@ contents:
         while true; do
             if [ "$SINGLE_STACK_IPV6" = true ]
             then
-                hostname=$(curl http://[fe80::a9fe:a9fe%25br-ex]:80/openstack/2012-08-10/meta_data.json | jq -re .name)
+                nic=$(ip -6 route get fe80::a9fe:a9fe | awk '{print $5}')
```

Comment:
Can this return more than one result? Do you perhaps want to limit it to the first result? Or better, loop over all of them until one succeeds.

Reply:
```
ip route get
              get a single route
              this command gets a single route to a destination and prints its contents exactly as the kernel sees it.
```
we are safe

Reply:
What if it returns the wrong route because multiple links have ipv6 link local addressing? Incidentally, this is the situation which makes specifying the nic as we do here necessary.

Reply:
ok this is unsafe. I tested it on a dual stack interface and it's broken.
I'll come up with something better.

Reply:
I made it more robust and added a validation.